main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -1644,10 +1645,25 @@ object Lowering {
   }
 
   /**
-    * The type of a channel which can transmit variables of type `tpe`
+    * The type of a channel which can transmit variables of type `tpe`.
     */
   private def mkChannelTpe(tpe: Type, loc: SourceLocation): Type = {
-    Type.Apply(Type.Cst(TypeConstructor.Enum(Enums.ChannelMpmc, Kind.Star ->: Kind.Star), loc), tpe, loc)
+    mkChannelTpe(tpe, Type.IO, loc)
+  }
+
+  /**
+    * The type of a channel which can transmit variables of type `tpe1` in region `tpe2`.
+    */
+  private def mkChannelTpe(tpe1: Type, tpe2: Type, loc: SourceLocation): Type = {
+    Type.Apply(Type.Apply(Types.ChannelMpmc, tpe1, loc), tpe2, loc)
+  }
+
+  /**
+    * Returns `(t1, t2)` where `tpe = Concurrent.Channel.Mpmc[t1, t2]`.
+    */
+  private def extractChannelTpe(tpe: Type): (Type, Type) = eraseAliases(tpe) match {
```

Comment:
This `eraseAliases` seems ad hoc, but I am just maintaining behaviour from before